common/lib/xmodule/xmodule/tests/test_capa_module.py

```diff
@@ -414,6 +414,100 @@ def test_showanswer_finished(self):
                                             graceperiod=self.two_day_delta_str)
         self.assertTrue(still_in_grace.answer_available())
 
+    def test_show_correctness_other(self):
+        """
+        Test that correctness is visible if show_correctness is not set to one of the values
+        from SHOW_CORRECTNESS constant.
+        """
+        show_correctness_empty = CapaFactory.create(show_correctness='')
+        self.assertTrue(show_correctness_empty.correctness_available())
+
+        show_correctness_other = CapaFactory.create(show_correctness='other-value')
+        self.assertTrue(show_correctness_other.correctness_available())
+
+    def test_show_correctness_default(self):
+        """
+        Test that correctness is visible by default.
+        """
+        problem = CapaFactory.create()
+        self.assertTrue(problem.correctness_available())
+
+    def test_show_correctness_never(self):
+        """
+        Test that correctness is hidden when show_correctness turned off.
+        """
+        problem = CapaFactory.create(show_correctness='never')
+        self.assertFalse(problem.correctness_available())
+
+    def test_show_correctness_closed(self):
+        """
+        Test that correctness is visible when learner is no longer allowed to submit answers,
+        and hidden if they are.
+        """
+        # Correctness visible after using up all attempts, even if due date in the future
+        used_all_attempts = CapaFactory.create(show_correctness='closed',
+                                               max_attempts="1",
+                                               attempts="1",
+                                               due=self.tomorrow_str)
+        self.assertTrue(used_all_attempts.correctness_available())
+
+        # Correctness visible if due date in the past
+        after_due_date = CapaFactory.create(show_correctness='closed',
+                                            max_attempts="1",
+                                            attempts="0",
+                                            due=self.yesterday_str)
+        self.assertTrue(after_due_date.correctness_available())
+
+        # Correctness not visible if attempts left
```

Comment:
@itsjeyd nit: I'd suggest splitting this cases into dedicated tests, or maybe a data driven test; the benefit is as uaual - this test will fail at the first error, so you won't know if further cases would pass or not (and it is useful sometimes)>